### PR TITLE
SDK-221 [NodeJS SDK] Fix bug at 'Migration' method

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-221](https://incountry.atlassian.net/browse/SDK-221)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

**Review by artemgolozubov — DISMISSED** [review]
LGTM

File: src/storage.ts [hunk 1]
```diff
@@ -293,18 +294,26 @@ class Storage {
     const findFilter = { ..._findFilter, version: { $not: currentSecretVersion } };
     const findOptions = { limit };
     const { records, meta, errors } = await this.find(countryCode, findFilter, findOptions, requestOptions);
-    if (records.length === 0 && errors && errors[0]) {
-      throw errors[0].error;
+    if (records.length === 0) {
+      if (errors && errors[0]) {
```
Comment (RommeroInCountry) [inline src/storage.ts:298]: let's have it the same as in other sdks - just return result with errors and 0 migrated. No need to throw anything here :)

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

File: src/storage.ts [hunk 2]
```diff
@@ -293,18 +294,22 @@ class Storage {
...
 
-    await this.batchWrite(countryCode, records, requestOptions);
-
-    return {
+    const result: MigrateResult = {
       meta: {
         migrated: meta.count,
         totalLeft: meta.total - meta.count,
       },
     };
```
Comment (RommeroInCountry) [inline src/storage.ts:306]: this one should be calculated differently. 
Don't use meta.count for calculating - use records.length. Errorful records are not migrated :)

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

**Review by RommeroInCountry — APPROVED** [review]
LGTM

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Pull request merged**